src/coreclr/jit/llvm.cpp

```diff
@@ -1996,11 +2068,17 @@ void Llvm::ConvertShadowStackLocalNode(GenTreeLclVarCommon* node)
             case GT_LCL_VAR:
                 indirOper = lclVar->TypeIs(TYP_STRUCT) ? GT_OBJ : GT_IND;
                 break;
+            case GT_LCL_VAR_ADDR:
```

Comment:
ok, I just reassigned `offset`, discarding the original value.

Reply:
I think it would be better to create the right one from the start. This offset will be the right one for all cases, `LCL_VAR/LCL_FLD/STORE_LCL_VAR/STORE_LCL_FLD`.

Reply:
I can do that with a tertiary : ` + (node->isLclField() ? node->AsLclFld()->GetLclOffs() : 0)`  ?

Reply:
`GetLclOffs` [is available](https://github.com/dotnet/runtimelab/blob/b135cf9b031395ffddb1b8479f1f9aef485e43ee/src/coreclr/jit/gentree.h#L3362) on `GenTreeLclVarCommon` nodes (and does that, yep).